zebra-network/src/peer/handshake.rs

```diff
@@ -227,7 +227,10 @@ pub async fn negotiate_version(
         timestamp,
         address_recv: (PeerServices::NODE_NETWORK, *addr),
         // TODO: detect external address (#1893)
-        address_from: (our_services, config.listen_addr),
+        address_from: (
+            our_services,
+            config.external_addr.unwrap_or(config.listen_addr),
```

Comment:
Can you turn this into an `external_addr()` method on `Config`?
(And make `Config.external_addr` private or `pub(crate)`.)

That way, we'll always get the external address the right way.

(We'll also want to use the external address in #1892 and similar tickets.)